### PR TITLE
Mrbbagging

### PR DESCRIPTION
**Pull request by plutasnyy**

### CODE REVIEW

File: benchmarks/mrbbagging/data/cleveland-sm.arff [hunk 1]
```diff
@@ -0,0 +1,300 @@
+@relation cleveland-wothu4
```
Comment (plutasnyy) [inline benchmarks/mrbbagging/data/cleveland-sm.arff]: `benchmarks/` is not the best directory for storing data ;)

File: multi_imbalance/ensemble/mrbbagging.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            classes_dict[i] = item
+        self.classifier_classes = classes_dict
+
+    def _count_votes(self, data):
+        voting_matrix = np.zeros((len(data), len(self.classes)))
+        for c in range(len(self.classifiers)):
+            classes = self.classifiers[c].predict(data)
+            probabilities = self.classifiers[c].predict_proba(data)
+            for i, cl in enumerate(classes):
+                idx = list(self.classifier_classes.keys())[list(self.classifier_classes.values()).index(cl)]
```
Comment (plutasnyy) [inline multi_imbalance/ensemble/mrbbagging.py]: This expression could be splitted

File: multi_imbalance/ensemble/mrbbagging.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    def _set_classes_dict(self, classes):
+        classes = list(classes)
+        classes_dict = {}
+        for i, item in enumerate(classes):
+            classes_dict[i] = item
+        self.classifier_classes = classes_dict
+
+    def _count_votes(self, data):
+        voting_matrix = np.zeros((len(data), len(self.classes)))
+        for c in range(len(self.classifiers)):
```
Comment (plutasnyy) [inline multi_imbalance/ensemble/mrbbagging.py]: `classifier_id` or `clf_id` is more descriptive and harder to confuse with class than `c`

File: multi_imbalance/ensemble/mrbbagging.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            self.classifiers[i] = la_list[i].fit(subset_x, subset_y)
+
+        return self
+
+    def _set_classes_dict(self, classes):
+        classes = list(classes)
+        classes_dict = {}
+        for i, item in enumerate(classes):
+            classes_dict[i] = item
+        self.classifier_classes = classes_dict
```
Comment (plutasnyy) [inline multi_imbalance/ensemble/mrbbagging.py]: `self.classifier_classess = dict(enumerate(classes))`

File: multi_imbalance/ensemble/mrbbagging.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    predict(data)
+        Predict classes for examples in data.
+    """
+    def __init__(self):
+        self.classifiers, self.classes, self.classifier_classes = dict(), dict(), dict()
+
+    def _group_data(self, x, y):
+        classes = set(y)
+        self.classes = {key: value for (key, value) in enumerate(classes)}
```
Comment (plutasnyy) [inline multi_imbalance/ensemble/mrbbagging.py]: `self.classes = dict(enumerate(classes))`
but is it really necessary to keep id:name map?

File: multi_imbalance/ensemble/mrbbagging.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        data = [[x[i], y[i]] for i in range(len(x))]
```
Comment (plutasnyy) [inline multi_imbalance/ensemble/mrbbagging.py]: Maybe using numpy arrays could be easier, eg. here `hstack`

File: multi_imbalance/ensemble/mrbbagging.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        grouped_data = dict()
+        for cl in classes:
+            assert cl is not None, "Missing class name"
+            grouped_data[cl] = list(filter(lambda d: d[1] == cl, data))
```
Comment (plutasnyy) [inline multi_imbalance/ensemble/mrbbagging.py]: nice!

File: benchmarks/mrbbagging/data/cleveland-sm.arff (hunk 1 above)
Comment (hancia) [reply benchmarks/mrbbagging/data/cleveland-sm.arff]: Followed the convention from SPIDER, can compress it if necessary.

File: multi_imbalance/ensemble/mrbbagging.py (hunk 5 above)
Comment (hancia) [reply multi_imbalance/ensemble/mrbbagging.py]: It's for prediction matrix purposes.

**Pull request merged**